res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1411 @@
+                DDJSR2.panels[1] = true;
+            } else if ((DDJSR2.panels[0] === true) && (DDJSR2.panels[1] === true)) {
+                DDJSR2.panels[0] = false;
+            } else if ((DDJSR2.panels[0] === false) && (DDJSR2.panels[1] === true)) {
+                DDJSR2.panels[1] = false;
+            }
+
+            engine.setValue("[Samplers]", "show_samplers", DDJSR2.panels[0]);
+            engine.setValue("[EffectRack1]", "show", DDJSR2.panels[1]);
+        }
+    };
+
+    this.shiftPanelSelectButton = function(channel, control, value, _status, _group) {
+        if (value) {
+            if ((DDJSR2.panels[0] === false) && (DDJSR2.panels[1] === false)) {
+                DDJSR2.panels[1] = true;
+            } else if ((DDJSR2.panels[1] === true) && (DDJSR2.panels[0] === false)) {
+                DDJSR2.panels[0] = true;
+            } else if ((DDJSR2.panels[0] === true) && (DDJSR2.panels[1] === true)) {
+                DDJSR2.panels[1] = false;
+            } else if ((DDJSR2.panels[1] === false) && (DDJSR2.panels[0] === true)) {
+                DDJSR2.panels[0] = false;
+            }
+
+            engine.setValue("[Samplers]", "show_samplers", DDJSR2.panels[0]);
+            engine.setValue("[EffectRack1]", "show", DDJSR2.panels[1]);
+        }
+    };
+};
+
+DDJSR2.MixerContainer.prototype = new components.ComponentContainer();
+
+DDJSR2.Channel = function(channelOffset) {
+    const deck = `[Channel${ channelOffset + 1 }]`;
+    this.pregain = new components.Pot({
+        // midi: [0xB0 + offset, 0x16],
+        group: deck,
+        inKey: "pregain",
+    });
+
+    this.eqKnob = [];
+    // EQ High, Mid and Low
+    for (let k = 1; k <= 3; k++) {
+        this.eqKnob[k] = new components.Pot({
+            group: `[EqualizerRack1_${  deck  }_Effect1]`,
+            inKey: `parameter${  k}`,
+        });
+    }
+
+    this.pfl = new components.Button({
+        midi: [0x90 + channelOffset, 0x54],
+        group: deck,
+        type: components.Button.prototype.types.toggle,
+        key: "pfl",
+    });
+
+    this.tapBPM = new components.Button({
+        midi: [0x90 + channelOffset, 0x68],
+        key: "bpm_tap"
+    });
+
+    this.filter = new components.Pot({
+        group: `[QuickEffectRack1_${ deck }]`,
+        inKey: "super1",
+    });
+
+
+    this.volume = new components.Pot({
+        inKey: "volume",
+        group: deck
+    });
+
+    // Master level is handled at hardware level for the SR2, same with Booth Monitor Level, Headphones Mixing, and Sampler Volume
+
+    let lastVuLevel = 0;
+    this.vuMeter = new components.Component({
+        outKey: "vu_meter",
+        group: deck,
+        midi: [0xB0, 0x02],
+        output: function(value, group, _control) {
+            // Remark: Only deck vu meters can be controlled! Master vu meter is handled by hardware!
+
+            value = parseInt(value * 0x76); //full level indicator: 0x7F
```

Comment:
the parseInt makes no sense here.
```suggestion
            value = value * 0x76; //full level indicator: 0x7F
```